### ISSUE TITLE
Control MousePosition fails using proj4 coordinate transformation and throws "TypeError: coordinates must be finite numbers"

### ISSUE THREAD

**Issue by anthonn** [openlayers/openlayers#15193]
I have an Angular application using openlayers 7.3.0. The map is in EPSG:3006.
There is a MousePosition control which shows the WGS84 coordinate in html. The control uses proj4 to do the transformation from EPSG:3006 to EPSG:4326.
This have been working for 5 years.

But now the application crashes when a user using a new IPAD running IOS 17 and using a stylus pen to the IPAD.

The following stacktrace comes from the error:

```
(webpack:///node_modules/proj4/lib/checkSanity.js:10:24) func: ja, code: throw new TypeError('coordinates must be finite numbers');
(webpack:///node_modules/proj4/lib/checkSanity.js:2:13) func: Fn, code: checkCoord(point.x);
(webpack:///node_modules/proj4/lib/transform.js:28:14) func: ua, code: checkSanity(point);
(webpack:///node_modules/proj4/lib/core.js:8:33) func: Fa, code: transformedArray = transform(from, to, coords, enforceAxis) || {x: NaN, y: NaN};
(webpack:///node_modules/ol/proj.js:720:30) func: , code: transformed = transform(coord);
(webpack:///node_modules/ol/proj.js:340:37) func: , code: const point = coordTransform(input.slice(i, i + dimension));
(webpack:///node_modules/ol/control/MousePosition.js:269:24) func: updateHTML_, code: this.transform_(coordinate, coordinate);
(webpack:///node_modules/ol/control/MousePosition.js:184:21) func: handleMouseMove, code: this.updateHTML_(map.getEventPixel(event));
(webpack:///node_modules/zone.js/fesm2015/zone.js:173:57) func: runTask, code: return this._zoneDelegate.invokeTask(this, task, applyThis, applyArgs);
(webpack:///node_modules/zone.js/fesm2015/zone.js:483:41) func: invokeTask, code: return task.zone.runTask(task, target, args);
(webpack:///node_modules/zone.js/fesm2015/zone.js:1631:24) func: he, code: task.invoke(task, target, [event]);
(webpack:///node_modules/zone.js/fesm2015/zone.js:1674:43) func: me, code: const err = invokeTask(copyTasks[i], target, event);
```

User-Agent for the device: 
```
crios-117.0.5938, iOS" "Mozilla/5.0 (iPad; CPU OS 17_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/117.0.5938.108 Mobile/15E148 Safari/604.1
```

I am guessing the problem is that somehow that the coordinate beeing sent from the function updateHTML_ in MousePosition to proj4 transform function could be something like this:
[Infinity, Infinity]

updateHTML_:
```
  updateHTML_(pixel) {
    let html = this.placeholder_;
    if (pixel && this.mapProjection_) {
      if (!this.transform_) {
        const projection = this.getProjection();
        if (projection) {
          this.transform_ = getTransformFromProjections(
            this.mapProjection_,
            projection
          );
        } else {
          this.transform_ = identityTransform;
        }
      }
      const map = this.getMap();
      const coordinate = map.getCoordinateFromPixelInternal(pixel);
      if (coordinate) {
        const userProjection = getUserProjection();
        if (userProjection) {
          this.transform_ = getTransformFromProjections(
            this.mapProjection_,
            userProjection
          );
        }
        this.transform_(coordinate, coordinate);
        if (this.wrapX_) {
          const projection =
            userProjection || this.getProjection() || this.mapProjection_;
          wrapX(coordinate, projection);
        }
        const coordinateFormat = this.getCoordinateFormat();
        if (coordinateFormat) {
          html = coordinateFormat(coordinate);
        } else {
          html = coordinate.toString();
        }
      }
    }
    if (!this.renderedHTML_ || html !== this.renderedHTML_) {
      this.element.innerHTML = html;
      this.renderedHTML_ = html;
    }
  }
```

**Comment by mike-000** [comment]
`map.getCoordinateFromPixelInternal(pixel)` should not return anything other than null or finite coordinates unless it was supplied with an invalid pixel, which is obtained using `map.getEventPixel(event)` https://github.com/openlayers/openlayers/blob/main/src/ol/Map.js#L800-L816  There may be an issue there with the event from your device.

**Comment by anthonn** [comment]
I agree that this probably could be a faulty pixel beeing reported from the device.
But could not Openlayers do a simple check to see that no faulty pixels are retrieved?

The issue was reported from a brand new Ipad, so it is a common device. It might be so that the Ipad software will fix the issue, but it would be nice if Openlayers were resilient agains this kind of problem.

**Comment by mike-000** [comment]
I think the issue will occur elsewhere in OpenLayers with pointermove events on that device, maybe in https://codesandbox.io/s/simple-forked-sd7pgs?file=/main.js for example, and each potential use would need to be checked as there is normally no need to suspect a pixel could be invalid.  If the current code is returning invalid values it would be better to debug the event and see if a meaningful pixel can be determined from the available properties.

**Comment by anthonn** [comment]
My user said it seems to work now after he changed a setting to off "Only Draw with Apple Pencil (When enabled, only Apple Pencil will draw. Your fingers will be used for scrolling instead)"

The behaviour with the setting on was that the pen seemed to change behaviour of the map when he lifted the pen, from ok behaviour, then to zooming and third, lift and down again, the map started rotating.
Since I have not seen it myself I cannot say if this was exactly the case, but maybe this info someday could help.

It would be really nice if openlayers could somehow detect when an ipad with a pen having the setting "Only Draw with Apple Pencil# to on. And maybe could filter out the input from the fingers/hand automatically.
But otherwise this issue seems to be solved for now on my end for now, until more users is having the same issue. 
Will see if I can find a way to detect if this is happening, but hard when not having an ipad.

**Comment by anthonn** [comment]
Unrelated to this issue I tried to make a longpress detection for selecting multiple features on touch event. The longpress was detected by monitoring the touchstart and that no touchend, touchmove (to far away) or touchcancel was detected for 500ms.
It worked good.
But first I tried stopPropagation() on the first pointerup event efter a detected longpress, in order to inhibit the "click" and "singleclick" event on the map. This made the touch (in chrome debug) zoom out and in, in very strange way. I guess the touchend event is important for the following mapbrowserevents.
I do not now if this might be related to the Apple pen issue, but maybe.

**Comment by anthonn** [comment]
I got som more information from my user now.

Apparently the setting "Only Draw with Apple Pencil" off, did not actually help. The most probably reason it now seems to work in my application is because som default interactions where turned off, before all were active:

```  javascript
    this.map = new olMap({
      view: view,
      controls: olControlDefaults({
        attribution: false,
        rotate: false,
      }),
      interactions: defaults({
        pinchRotate: false,
        altShiftDragRotate: false,
        keyboard: false,
        shiftDragZoom: false,
        doubleClickZoom: true,
        dragPan: true,
        mouseWheelZoom: true,
        pinchZoom: true,
        // zoomDelta: 1.5,
        // onFocusOnly: true,
        // zoomDuration: 0,
      }),
      pixelRatio: pixelRatio
    });
```
 
But in one of the examples on openlayers.org this still happens when the user uses a new Ipad with Apple Pen 2:
[https://www.icloud.com/photos/#0aaGDxUVu9YirTgJpK0N7DVYQ](https://www.icloud.com/photos/#0aaGDxUVu9YirTgJpK0N7DVYQ)

In the video the user says e uses the pen "normally" sometimes lifting and sometimes moving it along the screen. He also reloads the page occationally in the video.
